packages/core/src/fhir-deduplication/shared.ts

```diff
@@ -355,11 +363,42 @@ export function pickMostDescriptiveStatus<T extends string>(
   return status;
 }
 
+const classRanking = [
+  "ACUTE",
+  "IMP",
+  "NONAC",
+  "EMER",
+  "OBSENC",
+  "SS",
+  "AMB",
+  "HH",
+  "FIELD",
+  "VR",
+] as const;
+
+type EncounterClassCode = (typeof classRanking)[number];
+type EncounterClassCoding = Coding & { code?: EncounterClassCode };
+
+export function pickMostSevereClass(
+  class1: EncounterClassCoding | undefined,
+  class2: EncounterClassCoding | undefined
+): EncounterClassCoding | undefined {
+  if (class1 && class2 && class1.code && class2.code) {
+    const class1Rank = classRanking.indexOf(class1.code.toUpperCase() as EncounterClassCode);
+    const class2Rank = classRanking.indexOf(class2.code.toUpperCase() as EncounterClassCode);
+    return class1Rank < class2Rank ? class1 : class2;
```

Comment:
[This comment on the release PR](https://github.com/metriport/metriport/pull/4398#discussion_r2280957048) brings a good point:

<img width="774" height="551" alt="image" src="https://github.com/user-attachments/assets/65ab82f5-eecf-4b3c-9d88-d48f418bc85b" />